### PR TITLE
CB-11412 improve template implementation

### PR DESCRIPTION
**Pull request by carynbear**
Removed symlinking.

### CODE REVIEW

File: cordova-lib/spec-cordova/create.spec.js [hunk 1]
```diff
@@ -38,15 +37,17 @@ var configNormal = {
...
-        www: {
-          url: path.join(__dirname, 'fixtures', 'base'), // "create" should copy or link the www child of this dir and not the dir itself.
-          link: true
-        }
-      }
-    };
+};
+
+var configSubDirPkgJson = {
+            lib: {
```
Comment (stevengill) [inline cordova-lib/spec-cordova/create.spec.js]: nitpick: Look like lib obj here has an extra tab

File: cordova-lib/spec-cordova/create.spec.js [hunk 2]
```diff
@@ -121,6 +122,36 @@ describe('create end-to-end', function() {
...
+        // Check if top level dirs exist.
+        var dirs = ['hooks', 'platforms', 'plugins', 'www'];
+        dirs.forEach(function(d) {
+            expect(path.join(project, d)).toExist();
+        });
+        expect(path.join(project, 'hooks', 'README.md')).toExist();
+        
+        //index.js and template folder should not exist (inner files should be copied to the project folder)
+        expect(path.join(project, 'index.js')).not.toExist();
+        expect(path.join(project, 'template')).not.toExist();
```
Comment (stevengill) [inline cordova-lib/spec-cordova/create.spec.js]: Is this assuming the template folder is named `template`?

File: cordova-lib/spec-cordova/fixtures/templates/withsubdirectory_package_json/template/blah.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+blah
```
Comment (stevengill) [inline cordova-lib/spec-cordova/fixtures/templates/withsubdirectory_package_json/template/blah.txt]: Did you mean to include this file?

File: cordova-lib/src/cordova/create.js [hunk 4]
```diff
@@ -252,133 +249,116 @@ function create(dir, optionalId, optionalName, cfg, fetchOpt) {
...
+        shell.mkdir(dst);
+        shell.cp('-R', path.join(src, '*'), dst);
+    }
+}
+
+/**
+ * Copies template files, and directories into a Cordova project directory.
+ * If the template exists in a subdirectory everything is copied. 
+ * Otherwise package.json, RELEASENOTES.md, .git, NOTICE, LICENSE, COPYRIGHT, and .npmignore are not copied over.
+ * A template directory, and platform directory must be passed.
```
Comment (stevengill) [inline cordova-lib/src/cordova/create.js]: I think you meant project directory not platform directory in this comment

**Comment by stevengill** [comment]
LGTM!

File: cordova-lib/spec-cordova/create.spec.js (hunk 2 above)
Comment (carynbear) [inline cordova-lib/spec-cordova/create.spec.js]: correct

File: cordova-lib/spec-cordova/fixtures/templates/withsubdirectory_package_json/template/blah.txt (hunk 3 above)
Comment (carynbear) [inline cordova-lib/spec-cordova/fixtures/templates/withsubdirectory_package_json/template/blah.txt]: no

**Comment by carynbear** [comment]
@stevengill fixed up the little details

File: cordova-lib/src/cordova/create.js [hunk 5]
```diff
@@ -252,133 +249,116 @@ function create(dir, optionalId, optionalName, cfg, fetchOpt) {
...
         }
+
+        // Update package.json name and version fields.
+        if (fs.existsSync(path.join(dir, 'package.json'))) {
+            var pkgjson = require(path.resolve(dir, 'package.json'));
+            if (cfg.name) {
+                pkgjson.name = cfg.name.toLowerCase();
+            }
+            pkgjson.version = '1.0.0';
+            fs.writeFile(path.join(dir, 'package.json'), JSON.stringify(pkgjson), function (err) {
```
Comment (stevengill) [inline cordova-lib/src/cordova/create.js]: Lets maybe do the write operation as Sync and lets add spaces to the new package.json file. See https://github.com/apache/cordova-lib/blob/9f630271d9c02824c0031ea4cdc0e4955b53d9e2/cordova-lib/src/plugman/registry/manifest.js#L101

**Pull request closed without merging**